### PR TITLE
'built footer'

### PR DESCRIPTION
**Pull request by trisboyd**
The organization of the elements is kind of a mess, but the design left me little choice.  Each block is slightly different but instead of creating separate blocks I matched them as much as I could and passed in several props to create differences.

One thing I did fudge (which we can change if we want) is the margin underneath each block title is different and I evened them out to the relative average.  This seems more consistent to me, but I'm open to disagreement.

### CODE REVIEW

File: src/constants/footer.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+export const FooterData = {
+    blocks: [
+        {
+            title: "Let's get in touch",
+            text1: 'hello@acc.com',
```
Comment (rileydanejohnston) [inline src/constants/footer.js]: email is Hello@AwesomeContainers.com per final design (it says Hello@AwesomeContaners.com, I assume it should be 'containers')

File: src/constants/footer.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            text2: '+65 0123 4567',
```
Comment (rileydanejohnston) [inline src/constants/footer.js]: phone number has been removed

File: src/constants/footer.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            text3: 'Blk 35 Mandalay Road # 13-37',
```
Comment (rileydanejohnston) [inline src/constants/footer.js]: address has changed to 'Address: 71 UBI Road 1, #08-42,  Oxley Bizhub, Singapore 408732'

File: src/constants/footer.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            size: 'small',
+            margin: 'bottom'
+
+        },
+        {
+            title: 'Follow us',
+            img1: null,
+            img2: null,
+            link1: 'instagram',
+            link2: 'twitter',
```
Comment (rileydanejohnston) [inline src/constants/footer.js]: twitter has been removed

File: src/constants/footer.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            text5: 'Singapore 308215',
+            size: 'small',
+            margin: 'bottom'
+
+        },
+        {
+            title: 'Follow us',
+            img1: null,
+            img2: null,
+            link1: 'instagram',
```
Comment (rileydanejohnston) [inline src/constants/footer.js]: it says 'AwesomeContainers' instead of instagram now

File: src/constants/footer.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+            text1: 'We currently operate in:',
+            text2: 'Singapore',
+            text3: 'Planning to launch in 2022:',
+            text4: 'UK, EU, OSEAN, China, Vietnam, Malaysia,',
+            text5: 'Taiwan'
+        }
+    ],
+    links: [
+        'Awesome Container Compnay © 2022',
```
Comment (rileydanejohnston) [inline src/constants/footer.js]: 'Company' is misspelled, i see it's spelled this way on the design but we should fix it

File: src/components/footer/Footer.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
+import { Block, BlockText, BlockTitle, FooterBlocks, FooterBottom, FooterWrapper, ImgBackground, Link, LinkImg, LinkWrapper, TextBlock, TextLine, TextLineS, TextWrapper } from "./styledFooter"
```
Comment (rileydanejohnston) [inline src/components/footer/Footer.js]: we should remove the things we are not using like 'BlockText' in both this import statement and in styledFooter.js

File: src/components/footer/Footer.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+import { FooterData } from '../../constants/footer';
+import insta from '../../images/instagram.svg';
+import twitter from '../../images/twitter.svg';
+
+const {
+    blocks,
+    links
```
Comment (rileydanejohnston) [inline src/components/footer/Footer.js:6]: links isn't used

File: src/components/footer/Footer.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+                            </TextBlock>
+                        </Block>
+                    )
+                })}
+            </FooterBlocks>
+            <FooterBottom>
+                <TextLine
+                    $size={true}
+                    $right={true}>
+                    Awesome Container Compnay © 2022
```
Comment (rileydanejohnston) [inline src/components/footer/Footer.js]: Company > compnay

File: src/components/footer/Footer.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+const Footer = () => {
+
+    return (
+        <FooterWrapper>
+            <FooterBlocks>
+                {blocks.map((data) => {
+                    if (data.hasOwnProperty('text1')) {
+                        return (
+                            <Block key={blocks.indexOf(data)}>
```
Comment (rileydanejohnston) [inline src/components/footer/Footer.js:17]: key property is not necessary. We only need it when we use map with a ul element. My bad on this again

**Review by rileydanejohnston — CHANGES_REQUESTED** [review]
Overall, looks great! I few small changes per the new design and we are ready.

File: src/constants/footer.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+            // need instagram link for this element
+        },
+        {
+            title: 'Where we work',
+            text1: 'We currently operate in: \nSingapore',
+            text2: 'Planning to launch in 2022: \nUK, EU, OSEAN, China, Vietnam, Malaysia, \nTaiwan'
+        }
+    ],
+    links: [
+        'Awesome Container Company © 2022',
```
Comment (rileydanejohnston) [inline src/constants/footer.js]: Is this ever used? I notice the HTML still has this as well, but company is misspelled in the HTML

File: src/components/footer/Footer.js (hunk 9 above)
Comment (rileydanejohnston) [reply src/components/footer/Footer.js]: I recommend either putting the text in the HTML or in the const file. Right now there's both.

File: src/components/footer/Footer.js [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+                                </TextBlock>
+                                <TextBlock>
+                                    <TextLine $size={data.size}>{data.text2}</TextLine>
+                                </TextBlock>
+                            </Block>
+                        )
+                    }
+
+                    else return (
+                        <Block key={blocks.indexOf(data)}>
```
Comment (rileydanejohnston) [inline src/components/footer/Footer.js]: Key is unnecessary

File: src/components/footer/styledFooter.js [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+    font-style: normal;
+    font-weight: normal;
+    font-size: ${props => props.$size ? '14px' : '18px'};
+    line-height: ${props => props.$size ? '17px' : '21px'};
+    color: #FFFFFF;
+    margin-bottom: ${props => props.margin ? '10px' : null};
+    margin-right: ${props => props.$right ? '40px' : null};
+    white-space: pre-line;
+`
+export const LinkWrapper = styled.div`
```
Comment (rileydanejohnston) [inline src/components/footer/styledFooter.js:42]: the text should be aligned in the middle of the image, try adding 'align-items: center;'. I think that will do the trick

**Review by rileydanejohnston — CHANGES_REQUESTED** [review]
I think it's close. A few more changes and it will be ready

**Comment by trisboyd** [comment]
I don't know how I missed that the name is placed in the constants and hardcoded AND misspelled.  I missed that TWICE even though you told me the second time.  Thanks for being observant.

**Review by rileydanejohnston — APPROVED** [review]

**Pull request merged**